web-tester/src/main/java/com/axonivy/ivy/webtest/engine/WebAppFixture.java

```diff
@@ -0,0 +1,219 @@
+ * class Test {
+ *   {@literal @Test}
+ *   void test(WebAppFixture fixture) {
+ *     open(EngineUrl.create().toUrl());
+ *     $(By.id("sessionUserName")).shouldBe(Condition.exactText("Developer"));
+ *     fixture.login("test", "test");
+ *     $(By.id("sessionUserName")).shouldBe(Condition.exactText("test"));
+ *   }
+ * }
+ * </code></pre>
+ * @since 10.0.14
+ */
+public class WebAppFixture {
+
+  private static final String VAR_PATH = "variables";
+  private static final String CONFIG_PATH = "configs";
+
+  /**
+   * <p>Login to a user.
+   * This user will be used as long as the browser context isn't closed or the {@link #logout()} method is called.
+   * It uses the login form of the Dev-Workflow-UI.</p>
+   * <p>Example:
+   * <pre><code>
+   * {@literal @Test}
+   * void test(WebAppFixture fixture) {
+   *   open(EngineUrl.create().toUrl());
+   *   $(By.id("sessionUserName")).shouldBe(Condition.exactText("Developer"));
+   *   fixture.login("test", "test");
+   *   $(By.id("sessionUserName")).shouldBe(Condition.exactText("test"));
+   * }
+   * </code></pre>
+   * @param username the name of the user
+   * @param password the password of the user
+   */
+  public void login(String username, String password) {
+    var currentUrl = Selenide.webdriver().driver().url();
+    open(EngineUrl.create().app("").path("dev-workflow-ui/faces/login.xhtml").toUrl());
+    $(By.id("loginForm:userName")).shouldBe(visible).sendKeys(username);
+    $(By.id("loginForm:password")).shouldBe(visible).sendKeys(password);
+    $(By.id("loginForm:login")).shouldBe(visible).click();
+    $(By.id("loginForm:login")).shouldNotBe(visible);
+    if (!"about:blank".equals(currentUrl)) {
+      open(currentUrl);
+    }
+  }
+
+  /**
+   * <p>Logout the current user.
+   * This method logout the current user by using the Dev-Workflow-UI.</p>
+   * <p>Example:
+   * <pre><code>
+   * {@literal @Test}
+   * void test(WebAppFixture fixture) {
+   *   open(EngineUrl.create().toUrl());
+   *   $(By.id("sessionUserName")).shouldBe(Condition.exactText("Developer"));
+   *   fixture.logout();
+   *   $(By.id("sessionUserName")).shouldBe(Condition.exactText("test"));
+   * }
+   * </code></pre>
+   */
+  public void logout() {
+    var currentUrl = Selenide.webdriver().driver().url();
+    open(EngineUrl.create().app("").path("dev-workflow-ui/faces/home.xhtml").toUrl());
+    $(By.id("sessionUserName")).shouldBe(visible).click();
+    $(By.id("sessionLogoutBtn")).shouldBe(visible).click();
+    open(currentUrl);
+  }
+
+  /**
+   * <p>Set a variable to a given value.
+   * This variable will be set as long it is not been {@link #resetVar(String)}.</p>
+   * <p>Example:
+   * <pre><code>
+   * {@literal @Test}
+   * void test(WebAppFixture fixture) {
+   *   fixture.var("myVar", "myNewValue");
+   * }
+   * </code></pre>
+   */
+  public void var(String name, String value) {
+    try {
+      var url = configRestUrl(VAR_PATH + "/" + name) + "?value=" + value;
```

Comment:
hiho guys @ivy-lli and @ivy-rew 

Had a quick look into this implementation, because I needed to fix something in another repo.
I can imagine that `name` and `value` are not properly encoded - only a URIBuilder can do this properly (imo).

But maybe I don't see the full picture here.

Reply:
Ah yes this was still "open", forgot about this yesterday...
I will have a look at it 👍